components/url/lib.rs

```diff
@@ -163,6 +165,16 @@ impl ServoUrl {
 
 impl fmt::Display for ServoUrl {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        //changes for bug 22485
+        if self.0.to_string().as_bytes().len() > 40 {
+            let mut hasher = DefaultHasher::new();
+            let hashed_url : String = self.0.to_string();
+            (*hashed_url).hash(&mut hasher);
```

Comment:
We should be able to use `self.0.hash(&mut hasher)`, I believe.

Reply:
done !